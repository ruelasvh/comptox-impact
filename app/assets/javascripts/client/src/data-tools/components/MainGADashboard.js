/**
 * Created by Victor H. Ruelas-Rivera on 3/28/17.
 * US EPA National Center for Computational Toxicology
 */
import React, { PropTypes } from 'react'
import { Grid, Row, Col } from 'react-bootstrap'
import '../styles/maingadashboard.css'

class MainGADashboard extends React.Component {
    constructor(props) {
        super(props);
    }

    loadGAEmbedAPI() {
        (function(w,d,s,g,js,fs){
            g=w.gapi||(w.gapi={});g.analytics={q:[],ready:function(f){this.q.push(f);}};
            js=d.createElement(s);fs=d.getElementsByTagName(s)[0];
            js.src='https://apis.google.com/js/platform.js';
            fs.parentNode.insertBefore(js,fs);js.onload=function(){g.load('analytics');};
        }(window,document,'script'));
    }

    appendDashboard() {
        gapi.analytics.ready(function() {

            /**
             * Authorize the user immediately if the user has already granted access.
             * If no access has been created, render an authorize button inside the
             * element with the ID "embed-api-auth-container".
             */
            var CLIENT_ID = '793061702170-5egs8fbsdai74gocs0kflusn9e4b52hr.apps.googleusercontent.com';

            gapi.analytics.auth.authorize({
                container: 'embed-api-auth-container',
                clientid: CLIENT_ID,
                userInfoLabel: ''
            });


            // Chart for displaying page views
            var pageviewsChart = new gapi.analytics.googleCharts.DataChart({
                reportType: 'ga',
                query: {
                    ids: 'ga:69871570',
                    metrics: 'ga:pageviews',
                    dimensions: 'ga:yearMonth',
                    filters: 'ga:pagePath==epa.gov/chemical-research/downloadable-computational-toxicology-data',
                    'start-date': '2016-01-01',
                    'end-date': 'yesterday'
                },
                chart: {
                    container: 'pageviews-chart-container',
                    type: 'LINE',
                    options: {
                        width: '100%',
                        title: 'Page Views By Month And Year'
                    }
                }
            });

            pageviewsChart.on('error', function (e) {
                console.log('pageviewsChart error', e);
            });

            // Chart for displaying unique pages views
            var uniquePageviewsChart = new gapi.analytics.googleCharts.DataChart({
                reportType: 'ga',
                query: {
                    ids: 'ga:69871570',
                    metrics: 'ga:uniquePageviews',
                    dimensions: 'ga:yearMonth',
                    filters: 'ga:pagePath==epa.gov/chemical-research/downloadable-computational-toxicology-data',
                    'start-date': '2016-01-01',
                    'end-date': 'yesterday'
                },
                chart: {
                    container: 'unique-pageviews-chart-container',
                    type: 'LINE',
                    options: {
                        width: '100%',
                        title: 'Page Views By Month And Year'
                    }
                }
            });

            uniquePageviewsChart.on('error', function (e) {
                console.log('uniquePageviews error', e);
            });

            // Chart for displaying mobile vs desktop
            var mobileDesktopChart = new gapi.analytics.googleCharts.DataChart({
                reportType: 'ga',
                query: {
                    ids: 'ga:69871570',
                    metrics: 'ga:users',
                    dimensions: 'ga:deviceCategory',
                    filters: 'ga:pagePath==epa.gov/chemical-research/downloadable-computational-toxicology-data',
                    'start-date': '2016-01-01',
                    'end-date': 'yesterday'
                },
                chart: {
                    container: 'mobile-desktop-chart-container',
                    type: 'PIE',
                    options: {
                        width: '100%',
                        title: 'Device Category'
                    }
                }
            });

            mobileDesktopChart.on('error', function (e) {
                console.log('mobileDesktopChart error', e);
            });

            // Chart for displaying referral sites
            var referralSiteChart = new gapi.analytics.googleCharts.DataChart({
                reportType: 'ga',
                query: {
                    ids: 'ga:69871570',
                    metrics: 'ga:users,ga:newUsers',
                    dimensions: 'ga:fullReferrer',
                    filters: 'ga:fullReferrer!@direct;ga:fullReferrer!@anal;ga:pagePath==epa.gov/chemical-research/downloadable-computational-toxicology-data',
                    'start-date': '30daysAgo',
                    'end-date': 'yesterday'
                },
                chart: {
                    container: 'referral-site-chart-container',
                    type: 'TABLE',
                    options: {
                        width: '100%',
                        title: 'Referral Site'
                    }
                }
            });

            referralSiteChart.on('error', function (e) {
                console.log('referralSiteChart error', e);
            });

            // Chart for displaying specific events
            var eventsChart = new gapi.analytics.googleCharts.DataChart({
                reportType: 'ga',
                query: {
                    ids: 'ga:69871570',
                    metrics: 'ga:totalEvents',
                    dimensions: 'ga:eventLabel',
                    filters: 'ga:pagePath==epa.gov/chemical-research/downloadable-computational-toxicology-data',
                    'start-date': '2016-01-01',
                    'end-date': 'yesterday'
                },
                chart: {
                    container: 'events-chart-container',
                    type: 'PIE',
                    options: {
                        width: '100%',
                        title: 'Total Events By Event Label'
                    }
                }
            });

            eventsChart.on('error', function (e) {
                console.log('eventsChart error', e);
            });

            // Chart for displaying average time on page
            var avgTimeOnPageChart = new gapi.analytics.googleCharts.DataChart({
                reportType: 'ga',
                query: {
                    ids: 'ga:69871570',
                    metrics: 'ga:avgTimeOnPage',
                    dimensions: 'ga:pagePath',
                    filters: 'ga:pagePath==epa.gov/chemical-research/downloadable-computational-toxicology-data',
                    'start-date': '30daysAgo',
                    'end-date': 'yesterday'
                },
                chart: {
                    container: 'avg-time-on-page-chart-container',
                    type: 'TABLE',
                    options: {
                        width: '100%',
                        title: 'Average Time On Page'
                    }
                }
            });

            avgTimeOnPageChart.on('error', function (e) {
                console.log('avgTimeOnPageChart error', e);
            });

            // Load all charts
            gapi.analytics.auth.on('success', function (response) {
                // Hide the auth-button
                document.querySelector("#embed-api-auth-container").style.display='none';

                pageviewsChart.execute();
                uniquePageviewsChart.execute();
                mobileDesktopChart.execute();
                referralSiteChart.execute();
                eventsChart.execute();
                avgTimeOnPageChart.execute();
            });

        });
    }

    componentDidMount() {
        this.loadGAEmbedAPI();
        this.appendDashboard();
    }

    render() {
        return (
            <div id="charts-container">
                <Grid>
                    <Row>
                        <h2>CompTox Data</h2>
                        <div id="embed-api-auth-container"></div>
                        <div id="pageviews-chart-container"></div>
                        <div id="unique-pageviews-chart-container"></div>
                        <div id="mobile-desktop-chart-container"></div>
                        <div id="referral-site-chart-container"></div>
                        <div id="events-chart-container"></div>
                        <div id="avg-time-on-page-chart-container"></div>
                    </Row>
                </Grid>
            </div>
        )
    }
}

export default MainGADashboard;