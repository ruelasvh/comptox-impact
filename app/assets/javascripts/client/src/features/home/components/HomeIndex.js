/**
 * Created by Victor H. Ruelas-Rivera on 2/10/17.
 * US EPA National Center for Computational Toxicology
 */
import React, { PropTypes } from 'react';
import { Grid, Row, Col, Thumbnail} from 'react-bootstrap';
import { Link } from 'react-router';

import Client from '../../../Client';
import scientists from '../img/NCCT_Staff.jpg';
import publications from '../img/Publications.jpg';
import data_tools from '../img/Data_Tools.jpg';
import impact from '../img/Impact.jpg';
import '../styles/HomeIndex.css';

export default class HomeIndex extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            data: []
        };

        this.handleFetchData = this.handleFetchData.bind(this);
        this.handleCancelData = this.handleCancelData.bind(this);
    }

    handleFetchData() {
        Client.search(data => {
            this.setState({
                data: data
            });
        });
    }

    handleCancelData() {
        this.setState({
            data: []
        });
    }

    componentDidMount() {
        this.handleFetchData();
    }

    render() {
        let data = this.state.data;
        console.log(data);

        return (
            <div className="LandingMain-container">
                <Grid>
                    <Row>
                        <Col>
                            {/*<h1 style={{textAlign: 'center'}}>EPA Computational Toxicology Impact</h1>*/}
                            <h1 style={{textAlign: 'center'}}>{data.title}</h1>
                        </Col>
                    </Row>

                    <div className="LandingMain-header-text-space"></div>

                    <Row>
                        <Col>
                            <p style={{fontSize: 'large', textAlign: 'justify'}}>{data.intro}</p>
                        </Col>
                    </Row>

                    <div className="LandingMain-text-button-space"></div>

                    <div className="LandingMain-button-text">
                        <Row>
                            <Col xs={12} md={3}>
                                <Link to="/scientists">
                                    <Thumbnail src={scientists} alt="NCCT_Staff">
                                        <h3>Scientists</h3>
                                    </Thumbnail>
                                </Link>
                            </Col>
                            <Col xs={12} md={3}>
                                <a href="/publications">
                                    <Thumbnail src={publications} alt="Publications">
                                        <h3>Publications</h3>
                                    </Thumbnail>
                                </a>
                            </Col>
                            <Col xs={12} md={3}>
                                <a href="/data-tools">
                                    <Thumbnail src={data_tools} alt="Data_Tools">
                                        <h3>Data & Tools</h3>
                                    </Thumbnail>
                                </a>
                            </Col>
                            <Col xs={12} md={3}>
                                <a href="/impact">
                                    <Thumbnail src={impact} alt="Impact">
                                        <h3>Impact</h3>
                                    </Thumbnail>
                                </a>
                            </Col>
                        </Row>
                    </div>
                </Grid>
            </div>
        )
    }
}