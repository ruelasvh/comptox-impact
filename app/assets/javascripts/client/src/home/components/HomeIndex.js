/**
 * Created by Victor H. Ruelas-Rivera on 2/10/17.
 * US EPA National Center for Computational Toxicology
 */
import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Grid, Row, Col, Thumbnail} from 'react-bootstrap';
import { Link } from 'react-router';
import { fetchHomeIfNeeded } from '../actions';
import scientists from '../img/NCCT_Staff.jpg';
import publications from '../img/Publications.jpg';
import data_tools from '../img/Data_Tools.jpg';
import impact from '../img/Impact.jpg';
import '../styles/homeindex.css';

class HomeIndex extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        const { dispatch } = this.props;
        dispatch(fetchHomeIfNeeded());
    }

    render() {
        const { homeData } = this.props;
        return (
            <div className="LandingMain-container">
                <Grid>
                    <Row>
                        <Col>
                            {/*<h1 style={{textAlign: 'center'}}>EPA Computational Toxicology Impact</h1>*/}
                            <h1 style={{textAlign: 'center'}}>{homeData.title}</h1>
                        </Col>
                    </Row>

                    <div className="LandingMain-header-text-space"></div>

                    <Row>
                        <Col>
                            <p style={{fontSize: 'large', textAlign: 'justify'}}>{homeData.intro}</p>
                        </Col>
                    </Row>

                    <div className="LandingMain-text-button-space"></div>

                    <div className="LandingMain-button-text">
                        <Row>
                            <Col xs={12} md={3}>
                                <Link to="/scientists">
                                    <Thumbnail src={scientists} alt="NCCT_Staff">
                                        <h3>Scientists</h3>
                                    </Thumbnail>
                                </Link>
                            </Col>
                            <Col xs={12} md={3}>
                                <Link to="/publications">
                                    <Thumbnail src={publications} alt="Publications">
                                        <h3>Publications</h3>
                                    </Thumbnail>
                                </Link>
                            </Col>
                            <Col xs={12} md={3}>
                                <Link to="/data-tools">
                                    <Thumbnail src={data_tools} alt="Data_Tools">
                                        <h3>Data & Tools</h3>
                                    </Thumbnail>
                                </Link>
                            </Col>
                            <Col xs={12} md={3}>
                                <Link to="/impact">
                                    <Thumbnail src={impact} alt="Impact">
                                        <h3>Impact</h3>
                                    </Thumbnail>
                                </Link>
                            </Col>
                        </Row>
                    </div>
                </Grid>
            </div>
        )
    }
}

HomeIndex.propTypes = {};

function mapStateToProps(state) {
    const {
        isFetching,
        lastUpdated,
        homeData
    } = state.homeReducer || {
        isFetching: false,
        homeData: {}
    };
    return {
        isFetching,
        lastUpdated,
        homeData
    };
}

export default connect(mapStateToProps)(HomeIndex)