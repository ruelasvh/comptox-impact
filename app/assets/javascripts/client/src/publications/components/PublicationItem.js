/**
 * Created by vruelasr on 9/28/16.
 */
import React, { PropTypes } from 'react';
import $ from 'jquery';
import AltmetricWidget from './AltmetricWidget';
import PlumxWidget from './PlumxWidget';
import KudosWidget from './KudosWidget';
import { Glyphicon, Badge, Row, Col} from 'react-bootstrap';
import wgxpath from 'wgxpath';
import '../styles/publicationitem.css';


class PublicationItem extends React.Component {
    constructor() {
        super();
        this.state = {
            metrics: new Map()
        };
        // Bind get metrics method
        this.handleMetrics = this.handleMetrics.bind(this);
    }

    /* Get metrics from plumx widget and store in this component's state*/
    handleMetrics(thisComponent) {
        const id = thisComponent.props.id;
        let data = [];
        let m = new Map();

        $(document).ready(function () {
            // Simulate mouse enter element
            $("#plumx-publication-item-" + id).mouseenter(function () {
                data = [];
                m.clear();
            });
            // Simulate mouse leave element
            $("#plumx-publication-item-" + id).mouseleave(function () {

                const plumxMetricsUsage = getElementByXpath(".//*[@id='plumx-publication-item-"+id+"']/div/div/div/ul");
                // console.log('node', plumxMetricsUsage);

                if (plumxMetricsUsage) {
                    $("span", plumxMetricsUsage).each(function () {
                        data.push($(this).text());
                    });

                    for (var i = 0; i < data.length; i++) {
                        if (data[i] === "Abstract Views: ") {
                            m.set(data[i], data[i+1]);
                        } else if (data[i] === "Downloads: ") {
                            m.set(data[i], data[i+1]);
                        } else if (data[i] === "Citation Indexes: ") {
                            m.set(data[i], data[i+1]);
                        }
                    }
                }

                // If publication has doi, store metrics otherwise reset map
                if (!thisComponent.props.publication.doi) {
                    m.clear();
                }

                thisComponent.setState({ metrics: m }, console.log(thisComponent.state.metrics));
            });
            // Fire mouse hovering over element (does not work with out previous statements)
            $("#plumx-publication-item-" + id).trigger("mouseover");
            // Give time to collect the data
            setTimeout(function () {
                $("#plumx-publication-item-" + id).trigger('mouseleave');
            }, 100);
        });
    }

    componentDidMount() {
        this.handleMetrics(this);
    }

    render() {
        const { centerWide, publication, id } = this.props;
        const metrics = this.state.metrics;

        // Display information in their own badge
        const PlumMetrics = () => (
            <div className="publication-item-inline-metric">
                <Glyphicon glyph="glyphicon glyphicon-eye-open" />
                &nbsp;<strong>Abstract Views</strong>
                &nbsp;<Badge>{metrics.has("Abstract Views: ") ? metrics.get("Abstract Views: ") : "N/A"}</Badge>
                <div className="divider"/>
                <Glyphicon glyph="glyphicon glyphicon-book" />
                &nbsp;<strong>Citations</strong>
                &nbsp;<Badge>{metrics.has("Citation Indexes: ") ? metrics.get("Citation Indexes: ") : "N/A"}</Badge>
                <div className="divider"/>
                <Glyphicon glyph="glyphicon glyphicon-download-alt" />
                &nbsp;<strong>Downloads</strong> <Badge>{metrics.has("Downloads: ") ? metrics.get("Downloads: ") : "N/A"}</Badge>
            </div>
        );

        const Citation = () => (
            <div>
                <span dangerouslySetInnerHTML={{__html: publication.citation}} />
                <span>
                    {
                        publication.doi ?
                        <a href={`https://doi.org/${publication.doi}`} target="_blank"> doi:{publication.doi}</a> : ''
                    }
                </span>
            </div>
        );

        const Project = () => (
            <div dangerouslySetInnerHTML={{__html: publication.project}} />
        );

        return (
            <div className="publication-item">
                <Row>
                    <Col md={centerWide ? 7 : 9}>
                        <div className="publication-item-citation">
                            <Citation />
                        </div>
                        <div className={centerWide ? 'publication-item-show-counts' : 'publication-item-no-show-counts'}>
                            <PlumMetrics />
                        </div>
                    </Col>
                    <Col md={centerWide ? 2 : 0}>
                        <div className={centerWide ? 'publication-item-show-project' : 'publication-item-hide-project'}>
                            <Project />
                        </div>
                    </Col>
                    <Col md={1}>
                        <div
                            className={"publication-item-button " + (publication.doi ? "" : "publication-item-button-disabled")}
                            id={"altmetric-publication-item-" + id}
                            key={id} >
                            <AltmetricWidget doi={publication.doi ? publication.doi : "no-doi"} />
                        </div>
                    </Col>
                    <Col md={1}>
                        <div
                            className={"publication-item-button " + (publication.doi ? "" : "publication-item-button-disabled")}
                            id={"plumx-publication-item-" + id}
                            key={id} >
                            <PlumxWidget doi={publication.doi ? publication.doi : "xxxx/xxxxxx"} />
                        </div>
                    </Col>
                    <Col md={1}>
                        <div className={"publication-item-button " + (publication.doi ? "" : "publication-item-button-disabled")}>
                            <KudosWidget doi={publication.doi ? publication.doi : "no-doi"} />
                        </div>
                    </Col>
                </Row>
            </div>
        )
    }
}

PublicationItem.propTypes = {
    id: PropTypes.string.isRequired,
    publication: PropTypes.object.isRequired,
    centerWide: PropTypes.bool.isRequired
};

// Return element from XPath
function getElementByXpath(path) {
    wgxpath.install(window);
    return document.evaluate(path, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
}

export default PublicationItem;