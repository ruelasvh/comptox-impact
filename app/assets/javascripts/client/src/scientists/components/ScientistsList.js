/**
 * Created by Victor H. Ruelas-Rivera on 2/1/17.
 * US EPA National Center for Computational Toxicology
 */
import React, { PropTypes } from 'react';
import { Button, Pager, Glyphicon, Thumbnail, Grid, Breadcrumb, FormGroup, InputGroup, FormControl } from 'react-bootstrap';
import { Link } from 'react-router';
import Isotope from 'isotope-layout';

// components
import Scientist from './ScientistsListItem';
import '../styles/scientistsindex.css';


class ScientistsIndex extends React.Component {
    constructor(props) {
        super(props);

        // Convert to array
        const scientistsArr = Object.keys(this.props.scientists).map(key => this.props.scientists[key]);

        this.state = {
            isotope: null,
            sortValue: true,
            scientists: scientistsArr,
            currentlyDisplayed: scientistsArr,
            searchTerm: ''
        };

        // Bind event handlers
        this.handleClickSort = this.handleClickSort.bind(this);
        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleIsotope = this.handleIsotope.bind(this);
    }

    handleClickSort() {
        this.setState({ sortValue: !this.state.sortValue });
    }

    handleInputChange(event) {
        let newlyDisplayed = this.state.scientists.filter(
            (scientist) => {
                return (
                    scientist.firstName.toLowerCase().includes(event.target.value.toLowerCase()) ||
                    scientist.lastName.toLowerCase().includes(event.target.value.toLowerCase())
                    )
            }
            );

        this.setState({
            searchTerm: event.target.value,
            currentlyDisplayed: newlyDisplayed
        }, this.handleIsotope);
    }

    // Setup Isotope
    handleIsotope() {
        const grid = document.querySelector('.scientists-grid');
        this.setState({
            isotope: new Isotope( grid, {
                itemSelector: '.scientists-grid-item',
                layoutMode: 'masonry',
                masonry: {
                    columnWidth: 228
                },
                getSortData: {
                    name: '.last-name'

                }
            })
        });
    }

    componentDidMount() {
        this.handleIsotope();
    }


    render() {
        // Sorting icon
        const sortGlyph = !this.state.sortValue ? 'glyphicon glyphicon-sort-by-alphabet' : 'glyphicon glyphicon-sort-by-alphabet-alt';

        // List of scientists
        const ScientistsList = this.state.currentlyDisplayed.map(scientist => (
                        <div className="scientists-grid-item" key={scientist.scientistId}>
                            <Scientist scientist={scientist}/>
                        </div>
            ));

        return (
            <div className="scientists-container">
                <Grid>
                    <Breadcrumb>
                        <Breadcrumb.Item>
                            <Link to="/">Home</Link>
                        </Breadcrumb.Item>
                        <Breadcrumb.Item active>
                            Scientists
                        </Breadcrumb.Item>
                    </Breadcrumb>
                    <div className="scientists-controls-container">
                        <Button
                            bsStyle="primary"
                            bsClass="scientists-sort-button-style"
                            onClick={this.handleClickSort}>
                            <Glyphicon glyph={sortGlyph} /> Sort By Last Name
                        </Button>
                        <div className="scientists-search-form-style">
                            <FormGroup>
                                <InputGroup>
                                    <InputGroup.Addon>
                                        <Glyphicon glyph="glyphicon glyphicon-search"/>
                                    </InputGroup.Addon>
                                    <FormControl
                                        type="text"
                                        placeholder="Search Scientists"
                                        onChange={this.handleInputChange}/>
                                </InputGroup>
                            </FormGroup>
                        </div>
                    </div>

                    <div className="scientists-grid">
                        {ScientistsList}
                    </div>
                </Grid>
            </div>
        );
    }

    // When the DOM is rendered, let Isotope know what's changed
    componentDidUpdate() {
        if (this.state.isotope) {
            this.state.isotope.arrange(
                { sortBy: 'name', sortAscending: this.state.sortValue }
            );
        }

    }
}

export default ScientistsIndex;