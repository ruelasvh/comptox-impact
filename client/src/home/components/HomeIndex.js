/**
 * Created by Victor H. Ruelas-Rivera on 2/10/17.
 * US EPA National Center for Computational Toxicology
 */
import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Grid, Row, Col, Thumbnail} from 'react-bootstrap';
import { Link } from 'react-router';
import { fetchHomeDataIfNeeded } from '../actions';
import scientists from '../img/NCCT_Staff.jpg';
import publications from '../img/Publications.jpg';
import data_tools from '../img/Data_Tools.jpg';
import impact from '../img/Impact.jpg';
import '../styles/homeindex.css';

export class HomeIndex extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        const { dispatch } = this.props;
        dispatch(fetchHomeDataIfNeeded());
    }

    render() {
        const { homeData } = this.props;
        return (
            <div className="LandingMain-container">
                <Grid>
                    <Row>
                        <Col>
                            {/*<h1 style={{textAlign: 'center'}}>EPA Computational Toxicology Impact</h1>*/}
                            <h1 style={{textAlign: 'center'}}>{homeData.title}</h1>
                        </Col>
                    </Row>

                    <div className="LandingMain-header-text-space"></div>

                    <Row>
                        <Col>
                            <p style={{fontSize: 'large', textAlign: 'justify'}}>{homeData.intro}</p>
                        </Col>
                    </Row>

                    <div className="LandingMain-text-button-space"></div>

                    <div className="LandingMain-button-text">
                        <Row>
                            <Col xs={12} md={3}>
                                <Link to="/scientists">
                                    <Thumbnail src={scientists} alt="NCCT_Staff">
                                        <h3>Scientists <p><br/></p></h3>
                                    </Thumbnail>
                                </Link>
                            </Col>
                            <Col xs={12} md={3}>
                                <Link to="/publications-presentations">
                                    <Thumbnail src={publications} alt="Publications_Presentations">
                                        <h3>Publications and Presentations</h3>
                                    </Thumbnail>
                                </Link>
                            </Col>
                            <Col xs={12} md={3}>
                                <Link to="/data-tools">
                                    <Thumbnail src={data_tools} alt="Data_Tools">
                                        <h3>Data and Tools <p><br/></p></h3>
                                    </Thumbnail>
                                </Link>
                            </Col>
                            <Col xs={12} md={3}>
                                <Link to="/impact">
                                    <Thumbnail src={impact} alt="Impact">
                                        <h3>Impact <p><br/></p></h3>
                                    </Thumbnail>
                                </Link>
                            </Col>
                        </Row>
                    </div>
                </Grid>
            </div>
        )
    }
}

HomeIndex.propTypes = {};

function mapStateToProps(state) {
    const {
        isFetching,
        lastUpdated,
        homeData
    } = state.simpleHomeData || {
        isFetching: false,
        homeData: {}
    };
    return {
        isFetching,
        lastUpdated,
        homeData
    };
}

export default connect(mapStateToProps)(HomeIndex)