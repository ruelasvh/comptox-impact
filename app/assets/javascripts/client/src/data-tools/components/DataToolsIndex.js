/**
 * Created by Victor H. Ruelas-Rivera on 3/27/17.
 * US EPA National Center for Computational Toxicology
 */
import React, { PropTypes } from 'react'
import MainGADashboard from './MainGADashboard'

class DataToolsIndex extends React.Component {
    constructor(props) {
        super(props)
    }

    render() {
        return (<MainGADashboard/>)
    }
}

export default DataToolsIndex;