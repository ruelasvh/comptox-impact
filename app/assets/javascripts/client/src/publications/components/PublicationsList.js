/**
 * Created by Victor H. Ruelas-Rivera on 3/29/17.
 * US EPA National Center for Computational Toxicology
 */
import React, { PropTypes } from 'react';
import { Grid, Row, Col, FormControl, InputGroup, FormGroup, Glyphicon, Pagination } from 'react-bootstrap';
import PublicationItem from './PublicationItem';
import Isotope from 'isotope-layout';
import '../styles/publications.css';

export class PublicationsList extends React.Component {
    constructor(props) {
        super(props);

        const publicationsArr = Object.keys(this.props.publications).map(key => this.props.publications[key]);

        this.state = {
            activePage: 1,
            publications: publicationsArr,
            currentlyDisplayed: publicationsArr
        };

        //Bindings
        this.handleSelectPage = this.handleSelectPage.bind(this);
    }

    // To be used with the Pagination widget.
    handleSelectPage(eventKey) {
        this.setState({
            activePage: eventKey
        });
    }

    render() {
            const Publications = (
                <div>
                    {this.state.publications.map(function (publication) {
                        return (
                            <PublicationItem
                                id={publication.publication_id.toString()}
                                key={publication.publication_id}
                                publication={publication}
                                centerWide={true} />
                        );
                    })}
                </div>
            );


        return (
            <div className="publications-container">
                <Grid>
                    <Row bsClass="with-underheader-1">
                        <Col md={5}>
                            <h1>Publications</h1>
                        </Col>
                        <Col md={3}>
                            <div className="search-title" style={{paddingTop: '25px'}}>
                                <h4>Search NCCT Publications</h4>
                            </div>
                        </Col>
                        <Col md={4}>
                            <div className="search-box">
                                <FormGroup>
                                    <InputGroup>
                                        <InputGroup.Addon>
                                            <Glyphicon glyph="glyphicon glyphicon-search" />
                                        </InputGroup.Addon>
                                        <FormControl type="text" placeholder="Authors, Publication Name, DOI" />
                                    </InputGroup>
                                </FormGroup>
                            </div>
                        </Col>
                    </Row>


                    {/**
                     * Load up the drowpdown menus
                     **/}
                    {/*<div style={{paddingBottom: '5px'}}>*/}
                        {/*<strong>Filter by:</strong>*/}
                    {/*</div>*/}
                    {/*<Row>*/}
                        {/*<Col md={3}>*/}
                            {/*<div id="publication-year-filter-title" className="publications-filter-title">*/}
                                {/*<h4>Published Year</h4>*/}
                            {/*</div>*/}
                            {/*<div className="section">*/}
                                {/*<Select*/}
                                    {/*multi={true}*/}
                                    {/*simpleValue={false}*/}
                                    {/*disabled={false}*/}
                                    {/*value={this.state.filterValuesYears}*/}
                                    {/*placeholder="2009 - 2016"*/}
                                    {/*options={this.state.optionsYears}*/}
                                    {/*onChange={this.handleSelectChangeYears} />*/}
                            {/*</div>*/}
                        {/*</Col>*/}
                        {/*<Col md={3}>*/}
                            {/*<div id="publication-project-filter-title" className="publications-filter-title">*/}
                                {/*<h4>Research Project</h4>*/}
                            {/*</div>*/}
                            {/*<div className="section">*/}
                                {/*<Select*/}
                                    {/*multi={true}*/}
                                    {/*simpleValue={false}*/}
                                    {/*disabled={false}*/}
                                    {/*value={this.state.filterValuesProjects}*/}
                                    {/*placeholder="e.g. Rapid Exposure and Dose"*/}
                                    {/*options={this.state.optionsProjects}*/}
                                    {/*onChange={this.handleSelectChangeProjects} />*/}
                            {/*</div>*/}
                        {/*</Col>*/}
                        {/*<Col md={3}>*/}
                            {/*<div id="publication-author-filter-title" className="publications-filter-title">*/}
                                {/*<h4>Author</h4>*/}
                            {/*</div>*/}
                            {/*<div className="section">*/}
                                {/*<Select*/}
                                    {/*multi={true}*/}
                                    {/*simpleValue={false}*/}
                                    {/*disabled={false}*/}
                                    {/*value={this.state.filterValuesAuthors}*/}
                                    {/*placeholder="e.g. Richard, A."*/}
                                    {/*options={this.state.optionsAuthors}*/}
                                    {/*onChange={this.handleSelectChangeAuthors} />*/}
                            {/*</div>*/}
                        {/*</Col>*/}
                    {/*</Row>*/}

                    {/**
                     * Spacing between filter dropdowns and publications.
                     **/}
                    <div style={{padding: '25px'}}/>

                    {/**
                     * Publications go start here.
                     **/}
                    <Row bsClass="with-underheader-2">
                        <Col md={7}>
                            <strong>Publications</strong>
                        </Col>
                        <Col md={2}>
                            <strong>Research Project</strong>
                        </Col>
                        <Col md={1}>
                            <strong>Altmetric</strong>
                        </Col>
                        <Col md={1}>
                            <strong>PlumX</strong>
                        </Col>
                        <Col md={1}>
                            <strong>Kudos</strong>
                        </Col>
                    </Row>


                    <div className="publications-item-grid">
                        {Publications}
                    </div>

                    <div className="publications-pagination">
                        <Pagination
                            prev
                            next
                            ellipsis
                            boundaryLinks
                            items={20}
                            maxButtons={5}
                            activePage={this.state.activePage}
                            onSelect={this.handleSelectPage} />
                    </div>
                </Grid>
            </div>
        );
    }

}

PublicationsList.propTypes = {
    publications: PropTypes.object.isRequired
};


export default PublicationsList;