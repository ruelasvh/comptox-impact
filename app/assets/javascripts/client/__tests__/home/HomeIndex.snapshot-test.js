/**
 * Created by Victor H. Ruelas-Rivera on 1/30/17.
 * US EPA National Center for Computational Toxicology
 */
import React from 'react';
import { HomeIndex } from '../../src/home/components/HomeIndex';
import renderer from 'react-test-renderer';
import seed from '../../mocks/homeMock';

test('HomeIndex initial snapshot test', () => {
    const component = renderer.create(
        <HomeIndex homeData={seed} dispatch={()=>{}}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});