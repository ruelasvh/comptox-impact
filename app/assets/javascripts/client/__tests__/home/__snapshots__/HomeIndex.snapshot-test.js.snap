exports[`test HomeIndex initial snapshot test 1`] = `
<div
  className="LandingMain-container">
  <div
    className="container">
    <div
      className="row">
      <div
        className="">
        <h1
          style={
            Object {
              "textAlign": "center",
            }
          }>
          CompTox Impact Title
        </h1>
      </div>
    </div>
    <div
      className="LandingMain-header-text-space" />
    <div
      className="row">
      <div
        className="">
        <p
          style={
            Object {
              "fontSize": "large",
              "textAlign": "justify",
            }
          }>
          CompTox Impact showcases NCCT research metrics.
        </p>
      </div>
    </div>
    <div
      className="LandingMain-text-button-space" />
    <div
      className="LandingMain-button-text">
      <div
        className="row">
        <div
          className="col-md-3 col-xs-12">
          <a
            onClick={[Function]}
            style={Object {}}>
            <div
              className="thumbnail">
              <img
                alt="NCCT_Staff"
                src="test-file-stub" />
              <div
                className="caption">
                <h3>
                  Scientists
                </h3>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-md-3 col-xs-12">
          <a
            onClick={[Function]}
            style={Object {}}>
            <div
              className="thumbnail">
              <img
                alt="Publications"
                src="test-file-stub" />
              <div
                className="caption">
                <h3>
                  Publications
                </h3>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-md-3 col-xs-12">
          <a
            onClick={[Function]}
            style={Object {}}>
            <div
              className="thumbnail">
              <img
                alt="Data_Tools"
                src="test-file-stub" />
              <div
                className="caption">
                <h3>
                  Data & Tools
                </h3>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-md-3 col-xs-12">
          <a
            onClick={[Function]}
            style={Object {}}>
            <div
              className="thumbnail">
              <img
                alt="Impact"
                src="test-file-stub" />
              <div
                className="caption">
                <h3>
                  Impact
                </h3>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
